fast_pauli/cpp/include/__summed_pauli_op.hpp

```diff
@@ -613,6 +613,27 @@ template <std::floating_point T> struct SummedPauliOp
         }
     }
 
+    /**
+     * @brief Return the components of the SummedPauliOp as a vector of PauliOps
+     *
+     * @return std::vector<PauliOp<T>>
+     */
+    std::vector<PauliOp<T>> split() const
+    {
+        std::vector<std::complex<T>> op_coeffs(n_pauli_strings());
+        std::vector<PauliOp<T>> ops;
+        ops.reserve(n_operators());
+
+        for (size_t k = 0; k < n_operators(); k++)
+        {
+            for (size_t i = 0; i < n_pauli_strings(); i++)
+                op_coeffs[i] = coeffs(i, k);
+            ops.push_back(PauliOp<T>(op_coeffs, pauli_strings));
```

Comment:
Not that this is performance critical, but should we use `std::move` here or `emplace_back` instead of `push_back`?

Reply:
In this specific case a newly created PauliOp is an rvalue, so it's supposed to land into `void push_back(value_type&& __x)` which calls emplace_back with move on it. Basically no copying should occur

Reply:
Isn't that only if we have perfect forwarding, which requires a move constructor for `PauliOp` (which we don't have)?

Reply:
although we didn't define it explicitly but compiler is supposed to automatically generate move constructor and move assignment operator for PauliOp

Reply:
You're right, I forget that if we don't define a copy constructor, a copy assignment operator, a move assignment operator, or a destructor then the move constructor will get generated automatically